### ISSUE TITLE
`unneeded_parentheses_in_closure_argument` Not Registering as a Rule

### ISSUE THREAD

**Issue by merktassel** [realm/SwiftLint#1698]
I'm trying to use the `unneeded_parentheses_in_closure_argument` rule in order to prevent things like this from happening:

``` Swift
Alamofire.request("https://my.site/api", parameters: parameters).responseJSON { (response) in
    ...
}
```

I would expect to get a warning prompting me to remove the parenthesis around my closure parameters, to make the call look more like this:

``` Swift
Alamofire.request("https://my.site/api", parameters: parameters).responseJSON { response in
    ...
}
```

However, I don't. I ran `swiftlint rules` and was unable to find the specified rule listed:

```
Loading configuration from '.swiftlint.yml'
+-----------------------------------------+--------+-------------+------------------------+---------------------------------------------------------------------------------------------------------------------------+
| identifier                              | opt-in | correctable | enabled in your config | configuration                                                                                                             |
+-----------------------------------------+--------+-------------+------------------------+---------------------------------------------------------------------------------------------------------------------------+
| attributes                              | yes    | no          | yes                    | warning, always_on_same_line: ["@IBAction", "@NSManaged"], always_on_line_above: []                                       |
| class_delegate_protocol                 | no     | no          | yes                    | warning                                                                                                                   |
| closing_brace                           | no     | yes         | yes                    | warning                                                                                                                   |
| closure_end_indentation                 | yes    | no          | yes                    | warning                                                                                                                   |
| closure_parameter_position              | no     | no          | yes                    | warning                                                                                                                   |
| closure_spacing                         | yes    | yes         | yes                    | warning                                                                                                                   |
| colon                                   | no     | yes         | yes                    | warning, flexible_right_spacing: false, apply_to_dictionaries: true                                                       |
| comma                                   | no     | yes         | yes                    | warning                                                                                                                   |
| compiler_protocol_init                  | no     | no          | yes                    | warning                                                                                                                   |
| conditional_returns_on_newline          | yes    | no          | yes                    | warning                                                                                                                   |
| control_statement                       | no     | no          | yes                    | warning                                                                                                                   |
| custom_rules                            | no     | no          | yes                    | user-defined                                                                                                              |
| cyclomatic_complexity                   | no     | no          | yes                    | warning: 10, error: 20, ignores_case_statements: true                                                                     |
| discarded_notification_center_observer  | no     | no          | yes                    | warning                                                                                                                   |
| dynamic_inline                          | no     | no          | yes                    | error                                                                                                                     |
| empty_count                             | yes    | no          | yes                    | error                                                                                                                     |
| empty_enum_arguments                    | no     | yes         | yes                    | warning                                                                                                                   |
| empty_parameters                        | no     | yes         | yes                    | warning                                                                                                                   |
| empty_parentheses_with_trailing_closure | no     | yes         | yes                    | warning                                                                                                                   |
| explicit_init                           | yes    | yes         | yes                    | warning                                                                                                                   |
| explicit_top_level_acl                  | yes    | no          | no                     | warning                                                                                                                   |
| explicit_type_interface                 | yes    | no          | no                     | warning                                                                                                                   |
| extension_access_modifier               | yes    | no          | yes                    | warning                                                                                                                   |
| fatal_error_message                     | yes    | no          | yes                    | warning                                                                                                                   |
| file_header                             | yes    | no          | yes                    | warning, required_string: None, required_pattern: None, forbidden_string: None, forbidden_pattern: None                   |
| file_length                             | no     | no          | no                     | warning: 400, error: 1000                                                                                                 |
| first_where                             | yes    | no          | yes                    | warning                                                                                                                   |
| for_where                               | no     | no          | yes                    | warning                                                                                                                   |
| force_cast                              | no     | no          | no                     | error                                                                                                                     |
| force_try                               | no     | no          | no                     | error                                                                                                                     |
| force_unwrapping                        | yes    | no          | no                     | warning                                                                                                                   |
| function_body_length                    | no     | no          | no                     | warning: 40, error: 100                                                                                                   |
| function_parameter_count                | no     | no          | yes                    | warning: 5, error: 8                                                                                                      |
| generic_type_name                       | no     | no          | yes                    | (min_length) w/e: 1/0, (max_length) w/e: 20/1000, excluded: [], allowed_symbols: [], validates_start_with_lowercase: true |
| identifier_name                         | no     | no          | yes                    | (min_length) w/e: 3/2, (max_length) w/e: 40/60, excluded: [], allowed_symbols: [], validates_start_with_lowercase: true   |
| implicit_getter                         | no     | no          | yes                    | warning                                                                                                                   |
| implicit_return                         | yes    | yes         | yes                    | warning                                                                                                                   |
| implicitly_unwrapped_optional           | yes    | no          | no                     | warning, mode: allExceptIBOutlets                                                                                         |
| large_tuple                             | no     | no          | yes                    | warning: 2, error: 3                                                                                                      |
| leading_whitespace                      | no     | yes         | yes                    | warning                                                                                                                   |
| legacy_cggeometry_functions             | no     | yes         | yes                    | warning                                                                                                                   |
| legacy_constant                         | no     | yes         | yes                    | warning                                                                                                                   |
| legacy_constructor                      | no     | yes         | yes                    | warning                                                                                                                   |
| legacy_nsgeometry_functions             | no     | yes         | yes                    | warning                                                                                                                   |
| line_length                             | no     | no          | no                     | warning: 120, error: 200, ignores urls: false, ignores function declarations: false, ignores comments: false              |
| mark                                    | no     | yes         | yes                    | warning                                                                                                                   |
| multiline_parameters                    | yes    | no          | yes                    | warning                                                                                                                   |
| nesting                                 | no     | no          | yes                    | (type_level) w: 1, (statement_level) w: 5                                                                                 |
| nimble_operator                         | yes    | yes         | yes                    | warning                                                                                                                   |
| no_extension_access_modifier            | yes    | no          | yes                    | error                                                                                                                     |
| notification_center_detachment          | no     | no          | yes                    | warning                                                                                                                   |
| number_separator                        | yes    | yes         | yes                    | warning, minimum_length: 0                                                                                                |
| object_literal                          | yes    | no          | yes                    | warning, image_literal: true, color_literal: true                                                                         |
| opening_brace                           | no     | yes         | yes                    | warning                                                                                                                   |
| operator_usage_whitespace               | yes    | yes         | yes                    | warning                                                                                                                   |
| operator_whitespace                     | no     | no          | yes                    | warning                                                                                                                   |
| overridden_super_call                   | yes    | no          | yes                    | warning, excluded: [[]], included: [["*"]]                                                                                |
| private_outlet                          | yes    | no          | no                     | warning, allow_private_set: false                                                                                         |
| private_unit_test                       | no     | no          | yes                    | warning: XCTestCase                                                                                                       |
| prohibited_super_call                   | yes    | no          | yes                    | warning, excluded: [[]], included: [["*"]]                                                                                |
| protocol_property_accessors_order       | no     | yes         | yes                    | warning                                                                                                                   |
| redundant_discardable_let               | no     | yes         | yes                    | warning                                                                                                                   |
| redundant_nil_coalescing                | yes    | yes         | yes                    | warning                                                                                                                   |
| redundant_optional_initialization       | no     | yes         | yes                    | warning                                                                                                                   |
| redundant_string_enum_value             | no     | no          | yes                    | warning                                                                                                                   |
| redundant_void_return                   | no     | yes         | yes                    | warning                                                                                                                   |
| return_arrow_whitespace                 | no     | yes         | yes                    | warning                                                                                                                   |
| shorthand_operator                      | no     | no          | yes                    | error                                                                                                                     |
| sorted_imports                          | yes    | no          | yes                    | warning                                                                                                                   |
| statement_position                      | no     | yes         | yes                    | (statement_mode) default, (severity) warning                                                                              |
| switch_case_on_newline                  | yes    | no          | no                     | warning                                                                                                                   |
| syntactic_sugar                         | no     | no          | yes                    | warning                                                                                                                   |
| todo                                    | no     | no          | yes                    | warning                                                                                                                   |
| trailing_comma                          | no     | yes         | yes                    | warning, mandatory_comma: false                                                                                           |
| trailing_newline                        | no     | yes         | yes                    | warning                                                                                                                   |
| trailing_semicolon                      | no     | yes         | yes                    | warning                                                                                                                   |
| trailing_whitespace                     | no     | yes         | yes                    | warning, ignores_empty_lines: false, ignores_comments: true                                                               |
| type_body_length                        | no     | no          | yes                    | warning: 200, error: 350                                                                                                  |
| type_name                               | no     | no          | no                     | (min_length) w/e: 3/0, (max_length) w/e: 40/1000, excluded: [], allowed_symbols: [], validates_start_with_lowercase: true |
| unused_closure_parameter                | no     | yes         | yes                    | warning                                                                                                                   |
| unused_enumerated                       | no     | no          | yes                    | warning                                                                                                                   |
| unused_optional_binding                 | no     | no          | yes                    | warning, ignore_optional_try: false                                                                                       |
| valid_ibinspectable                     | no     | no          | yes                    | warning                                                                                                                   |
| vertical_parameter_alignment            | no     | no          | yes                    | warning                                                                                                                   |
| vertical_parameter_alignment_on_call    | yes    | no          | yes                    | warning                                                                                                                   |
| vertical_whitespace                     | no     | yes         | yes                    | warning, max_empty_lines: 1                                                                                               |
| void_return                             | no     | yes         | yes                    | warning                                                                                                                   |
| weak_delegate                           | no     | no          | yes                    | warning                                                                                                                   |
+-----------------------------------------+--------+-------------+------------------------+---------------------------------------------------------------------------------------------------------------------------+
```

`swiftlint lint` returns the following:

```
Loading configuration from '.swiftlint.yml'
Linting Swift files in current working directory
Linting 'fileName1.swift' (1/69)
Linting 'fileName2.swift' (2/69)
...
Linting 'fileName69.swift' (69/69)
Done linting! Found 0 violations, 0 serious in 69 files.
```

My configuration file (`.swiftlint.yml`) looks like this:

``` YAML
cyclomatic_complexity:
    ignores_case_statements: true

excluded:
    - Pods

whitelist_rules:
    - attributes
    - class_delegate_protocol
    - closing_brace
    - closure_end_indentation
    - closure_parameter_position
    - colon
    - comma
    - compiler_protocol_init
    - conditional_returns_on_newline
    - control_statement
    - closure_spacing
    - custom_rules
    - cyclomatic_complexity
    - discarded_notification_center_observer
    - dynamic_inline
    - empty_count
    - empty_enum_arguments
    - empty_parameters
    - empty_parentheses_with_trailing_closure
    - explicit_init
#    - explicit_top_level_acl
#    - explicit_type_interface
    - extension_access_modifier
    - fatal_error_message
    - file_header
#    - file_length
    - first_where
    - for_where
#    - force_cast
#    - force_try
#    - force_unwrapping
#    - function_body_length
    - function_parameter_count
    - generic_type_name
    - identifier_name
    - implicit_getter
    - implicit_return
#    - implicitly_unwrapped_optional
    - large_tuple
    - leading_whitespace
    - legacy_cggeometry_functions
    - legacy_constant
    - legacy_constructor
    - legacy_nsgeometry_functions
    - let_var_whitespace
#    - line_length
    - mark
    - multiline_parameters
    - nesting
    - nimble_operator
    - no_extension_access_modifier
    - notification_center_detachment
    - number_separator
    - object_literal
    - opening_brace
    - operator_usage_whitespace
    - operator_whitespace
    - operator_function_whitespace
    - overridden_super_call
#    - private_outlet
    - private_over_fileprivate
    - private_unit_test
    - prohibited_super_call
    - protocol_property_accessors_order
    - redundant_discardable_let
    - redundant_nil_coalescing
    - redundant_optional_initialization
    - redundant_string_enum_value
    - redundant_void_return
    - return_arrow_whitespace
    - shorthand_operator
    - sorted_imports
    - statement_position
    - strict_fileprivate
#    - switch_case_on_newline
    - syntactic_sugar
    - todo
    - trailing_comma
    - trailing_newline
    - trailing_semicolon
    - trailing_whitespace
    - type_body_length
#    - type_name
    - unneeded_parentheses_in_closure_argument
    - unused_closure_parameter
    - unused_enumerated
    - unused_optional_binding
    - valid_ibinspectable
    - vertical_parameter_alignment
    - vertical_parameter_alignment_on_call
    - vertical_whitespace
    - void_return
    - weak_delegate
```

All other rules whitelisted seem to be working. I've read [the rule documentation](https://github.com/realm/SwiftLint/blob/master/Rules.md#unneeded-parentheses-in-closure-argument) and don't see any problem with how I'm trying to use the rule, unless I misunderstood something. My `swiftlint version` is `0.20.0`, and I'm using Cocoapods for the installation. Any ideas what could be going on?

**Comment by jpsim** [comment]
That rule was just merged to `master` yesterday in #1684, so it won't be included in 0.20.0.

**Comment by merktassel** [comment]
Is it available in the latest release? I just ran `pod update SwiftLint` and now `swiftlint version` outputs `0.20.1`, but I still don't see the rule listed by `swiftlint rules`.

**Comment by jpsim** [comment]
No the latest release is 0.20.1, which was made ~3 weeks ago.
